x/observer/types/observer_set.go

```diff
@@ -14,14 +15,25 @@ func (m *ObserverSet) LenUint() uint64 {
 	return uint64(len(m.ObserverList))
 }
 
-// Validate observer mapper contains an existing chain
+// Validate observer set verifies that the observer set is valid
+// - All observer addresses are valid
+// - No duplicate observer addresses
 func (m *ObserverSet) Validate() error {
```

Comment:
Validate() can be refactores to achieve O(n) complexity.
Also it would be nice to add `ErrInvalidObserverAddress` as a custom error.
```suggestion
func (m *ObserverSet) Validate() error {
	observers := make(map[string]struct{})

	for _, observerAddress := range m.ObserverList {
		// Check address is valid.
		_, err := sdk.AccAddressFromBech32(observerAddress)
		if err != nil {
			return errors.Wrapf(ErrInvalidObserverAddress, "invalid observer address: %s", observerAddress)
		}
		
		// Check for duplicate observers.
		if _, ok := observers[observerAddress]; ok {
			return errors.Wrapf(ErrDuplicateObserver, "observer %s", observerAddress)
		}

		// Update observers map.
		observers[observerAddress] = struct{}{}
	}

	return nil
}
```